.gitignore

```diff
@@ -9,3 +9,5 @@
 *~
 .gdb_history
 *.out
+build/test/*
```

Comment:
Ah yeah I don't like including the build folder and I just get frustrated when I delete the build folder to do a clean build and then the .gitignore is deleted. If we have to keep the empty build directory in the repo then I'd prefer to keep these lines in the gitignore

Reply:
I would use `build/*` then, instead of `build/test/*`, in this `.gitignore`.

Reply:
I'm not sure what difference `build/*` would have over simply `build` or `build/`, except that I think that `build/.gitignore` would be ignored with `build` or `build/`, but not with `build/*`, which is what we want (so that `build/.gitignore` changes are still tracked).

Reply:
So I guess that I'm saying that `build/*` should be used. That would cover your case while not including `build/.gitignore`, because `*` does not include filenames beginning with `.`.